### PR TITLE
Add scipy global optimizers

### PR DESCRIPTION
**Pull request by baharcos**

### CODE REVIEW

File: tests/optimization/test_many_algorithms.py [hunk 1]
```diff
@@ -70,7 +69,7 @@ def test_algorithm_on_sum_of_squares_with_binding_bounds(algorithm):
 )
 
 
-@pytest.mark.skipif(sys.platform != "linux", reason=skip_msg)
+# @pytest.mark.skipif(sys.platform != "linux", reason=skip_msg)
```
Comment (timmens) [inline tests/optimization/test_many_algorithms.py:72]:
```suggestion
@pytest.mark.skipif(sys.platform != "linux", reason=skip_msg)
```

**Comment by janosg** [comment]
Hi @baharcos and @timmens. Thanks a lot for this PR. I went over all optimizers and added a few finishing touches:
- renaming some of the options to be more consistent with the rest of estimagic
- use estimagics batch_evaluators for parallelization
- pass a few more things through (e.g. everything that uses a local algorithm passes derivatives through to the local algorithm).

From my side we could merge. Let me know if this is ok for you.

File: docs/source/algorithms.md [hunk 2]
```diff
@@ -565,6 +565,222 @@ you install estimagic.
...
+```{eval-rst}
+.. dropdown::  scipy_shgo
+
+    .. code-block::
+
+        "scipy_shgo"
+
+    Find the global minimum of a fuction using simplicial homology global optimization.
+
+    Note: scipy_shgo requires the constraints' codomain to be a scalar.
```
Comment (timmens) [inline docs/source/algorithms.md:698]: This is not true anymore. We transform vector-valued constraints into a list of scalar-valued constraints for this algorithm.

Comment (timmens) [inline docs/source/algorithms.md:698]:
```suggestion
```

**Review by timmens — APPROVED** [review]
Looks good to me.

**Review by janosg — APPROVED** [review]

**Pull request merged**